assets/apps/components/src/Controls/ButtonAppearance.js

```diff
@@ -104,6 +151,92 @@ const ButtonAppearance = ({ label, value, onChange, noHover, defaultVals }) => {
 									);
 								}
 							)}
+							{/*
+							 * Shadow toggle for Button Appearance
+							 */}
+							<hr />
+							<PanelRow key={`shadow_${optionType}`}>
+								<Toggle
+									label={
+										__('Button', 'neve') +
+										' ' +
+										__('Shadow', 'neve')
+									}
+									checked={value[shadowToggle]}
+									onChange={(nextValue) => {
+										onChange(shadowToggle, nextValue);
+									}}
+								/>
+							</PanelRow>
+							{/*
+							 * Shadow Controls for Button Appearance
+							 */}
+							{value[shadowToggle] &&
+								Object.keys(
+									shadowSettings[optionType].controls
+								).map((controlSlug, controlIndex) => {
+									const {
+										label: shadowControlLabel,
+										type: shadowControlType,
+										...controlArgs
+									} = shadowSettings[optionType].controls[
+										controlSlug
+									];
+									return (
+										<PanelRow key={controlIndex}>
+											{shadowControlType === 'color' && (
+												<ColorControl
+													label={shadowControlLabel}
+													defaultValue={
+														defaultVals[
+															shadowColor
+														] || null
+													}
+													selectedColor={
+														value[shadowColor]
+													}
+													onChange={(nextColor) => {
+														onChange(
+															shadowColor,
+															nextColor
+														);
+													}}
+												/>
+											)}
+											{shadowControlType === 'range' && (
+												<Range
+													label={shadowControlLabel}
+													defaultVal={
+														defaultVals[
+															shadowProperties
+														][controlSlug]
```

Comment:
We should either check each key individually here or add a default I mentioned in the main review comment.
```suggestion
														defaultVals[shadowProperties] && defaultVals[shadowProperties][controlSlug] ? defaultVals[shadowProperties][controlSlug] : null // or what other value you want to default to. 
```

The same thing should be done for all other nested values accessed with variable keys.